DUI3-DX/Connectors/Rhino/Speckle.Connectors.Rhino7/Operations/Send/BaseObjectSenderToServer.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Speckle.Connectors.Utils.Operations;
+using Speckle.Core.Api;
+using Speckle.Core.Credentials;
+using Speckle.Core.Models;
+using Speckle.Core.Transports;
+
+namespace Speckle.Connectors.Rhino7.Operations.Send;
+
+/// <summary>
+/// Default implementation of the <see cref="IBaseObjectSender"/> which takes a <see cref="Base"/> and sends
+/// it to a server described by the parameters in the <see cref="Send"/> method
+/// </summary>
+internal sealed class BaseObjectSenderToServer : IBaseObjectSender
+{
+  private readonly Func<Account, string, ITransport> _transportFactory;
+
+  public BaseObjectSenderToServer(Func<Account, string, ITransport> transportFactory)
+  {
+    _transportFactory = transportFactory;
+  }
+
+  public async Task<string> Send(
+    Base commitObject,
+    string accountId,
+    string projectId,
+    string modelId,
+    Action<string, double?>? onOperationProgressed = null,
+    CancellationToken ct = default
+  )
+  {
+    ct.ThrowIfCancellationRequested();
+
+    onOperationProgressed?.Invoke("Uploading...", null);
+
+    // TODO: FETCHING ACCOUNTS BY ID ONLY IS UNSAFE
```

Comment:
Why is it unsafe? I think this is giving us the behaviour we want? I think the list of matching IDs is ordered with NWA/migrated accounts coming first?

Reply:
This was my comment. After our last changes it is "less unsafe" but still unsafe to assume there will only be 1 account with the same id at any given time.

Since we now ensure the "upgraded account" is the one returned first, this is less of an issue now but still I would very much push for

Filter accounts by server -> then pick by ID

This ensures we 100% pick the right account always.

The problem is that it doesn't seem like the modelCardId holds this data at all